fastlane/lib/fastlane/actions/notarize.rb

```diff
@@ -177,7 +213,15 @@ def self.available_options
                                        description: 'Whether to log requests',
                                        optional: true,
                                        default_value: false,
-                                       type: Boolean)
+                                       type: Boolean),
+          FastlaneCore::ConfigItem.new(key: :api_key_path,
+                                       env_name: 'FL_NOTARIZE_API_KEY_PATH',
+                                       description: 'Path to AppStore Connect API key',
+                                       optional: true,
+                                       is_string: true,
```

Comment:
we could add `conflicting_options: [:username]` and vice versa.

Reply:
Made optional and conflicting!

No logic is necessary to handle the case none of them is there, there will always be at least the `username` from its default value:
https://github.com/fastlane/fastlane/blob/afd4540ea9fdbff20813fc32acd60d6e7febf7bd/fastlane/lib/fastlane/actions/notarize.rb#L135-L136